pytorch_to_returnn/torch/_C.py

```diff
@@ -122,17 +123,46 @@ class Size(tuple):  # type: Tuple[SizeValue, ...]
 
 class SizeValue(int):
   """
-  We extend this, to store extra information, e.g. such that this reflects the batch-dim.
+  We extend this, to store extra information, e.g. corresponding RETURNN dim tags.
   """
-  is_batch_dim: bool = False
-  merged_dims: list[SizeValue] = []
+  def __new__(cls, x, dim_tag: Optional[Dim] = None, merged_dims: Optional[List[SizeValue]] = None):
+    res = super(SizeValue, cls).__new__(cls, x)
+    res.dim_tag = dim_tag or Dim(dimension=x, description="static_dim")
+    res.merged_dims = merged_dims or []
+    return res
+
+  @property
+  def is_batch_dim(self):
+    if self.dim_tag is None:
+      return False
+    return self.dim_tag.is_batch_dim()
 
   def __repr__(self):
     res = super(SizeValue, self).__repr__()
     if self.is_batch_dim:
       res = f"Batch({res})"
     return res
 
+  def __mul__(self, other):
+    if not isinstance(other, int):  # e.g. a list
```

Comment:
Do you really get this case? I wonder if we should do sth more clever, or maybe forbid it, because likely whatever this is used for, it would be wrong because it assumes a static batch dim.

Reply:
Only the "opposite" case in `__rmul__` occurred in the tests ([here](https://github.com/rwth-i6/pytorch-to-returnn/blob/c0b77753d908a3a5a125662da575012b6ae10cba/pytorch_to_returnn/torch/nn/utils/rnn.py#L43)). I added it in `__mul__` for consistency.

Reply:
We should maybe replace the `lengths = [time_dim] * batch_dim` by `fill` or some other construct. Because this is wrong.

Reply:
Yes, that's true. Should we just forbid this case and open an issue for the usage in `pack_sequence`? However, then the test case currently fails.

Reply:
Just directly fix `pack_sequence` in this PR as well. This should be a very simple fix.

Reply:
What exactly do you mean by `fill(length, [batch_dim])`? Something like this?
```
lengths.zeros(batch_dim)
lengths.fill_(time_dim)
```

Reply:
I meant what `tf.fill` is in TF. It looks like this is `torch.full`. I.e. `torch.full([batch_dim], time_dim)`. That will give you a tensor of shape [batch_dim] with entries time_dim everywhere. Exactly like `[time_dim] * batch_dim` but as a tensor.

Reply:
This change for `pack_sequence` is still needed, right?

Reply:
Yes, it is.